app/models/redirect.rb

```diff
@@ -50,20 +50,16 @@ def paths
   end
 
   def build_url_for path
-    URI.parse("http://" + strip_protocol_from_path(path))
-  end
-
-  def strip_protocol_from_path path
-    path.strip.gsub(%r{https*://}, "")
+    URI.parse path
   end
 
   def groomed_path_or_url url
-    url_pieces = []
-    url_pieces << url.host unless url.host =~ %r{crimethinc.com|cwc.im}
-    url_pieces << url.path
-    url_pieces << "?" + url.query    if url.query.present?
-    url_pieces << "#" + url.fragment if url.fragment.present?
-    url_pieces.join
+      url_pieces = []
+      url_pieces << "#{url.scheme}://#{url.host}" unless url.host =~ %r{crimethinc.com|cwc.im} or not url.host.present?
```

Comment:
the `unless` + `or not` combo is confusing me.

I think using `not something.present?` is the same as `something.blank?`

doing that substitution results in:
```ruby
url_pieces << "#{url.scheme}://#{url.host}" unless url.host =~ %r{crimethinc.com|cwc.im} or url.host.blank? 
```
now when I read this, I assume it is saying 

> add this string to `url_pieces` unless either one of these conditions is true:
>  1. it is a crimethinc url 
>  2. url_host is `nil`

if that is what it is supposed to do, I'd prefer:
```ruby
url_pieces << "#{url.scheme}://#{url.host}" unless crimethinc_url?

#...

def crimethinc_url?
  url.host.blank? || url.host =~ %r{crimethinc.com|cwc.im} 
end
```

Reply:
It's stupid since I didn't know about the `blank?` method, next time I'll know. For the `crimethinc_urls?` method, it's really cleaner that way !